tracer/src/Datadog.Trace.ClrProfiler.Native/calltarget_tokens.cpp

```diff
@@ -884,7 +885,9 @@ HRESULT CallTargetTokens::WriteBeginMethod(void* rewriterWrapperPtr, mdTypeRef i
     ILRewriterWrapper* rewriterWrapper = (ILRewriterWrapper*) rewriterWrapperPtr;
     ModuleMetadata* module_metadata = GetMetadata();
 
-    auto numArguments = (int) methodArguments.size();
+    auto numArgsToLoad = argsToLoad.size();
+    bool useArgToLoad = numArgsToLoad > 0;
+    auto numArguments = useArgToLoad ? (int) numArgsToLoad : (int) methodArguments.size();
```

Comment:
Again, just to clarify, changing this doesn't mean we accidentally select the wrong overload, it just means we don't pass those arguments to our instrumentation?

Reply:
that's right in this part of the code we are already rewriting the method, not selecting it.